core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -979,6 +1007,14 @@ private Void visitScanFilterAndProjectInfo(
                     double filtered = 100.0d * (nodeStats.getPlanNodeInputPositions() - nodeStats.getPlanNodeOutputPositions()) / nodeStats.getPlanNodeInputPositions();
                     nodeOutput.appendDetailsLine(", Filtered: %s%%", formatDouble(filtered));
                 }
+                List<DynamicFilterDomainStats> collectedDomainStats = dynamicFilters.stream()
```

Comment:
I think DF reporting can get quite large. IMO we should print it as part of `VERBOSE` flag.

Reply:
Each DF by itself won't be big as we're printing only first and last value of domain. Even with multiple DFs per FilterNode it should be comparable to Estimates printing
e.g.:
```
ScanFilterProject[table = hive:default:lineitem, grouped = false, filterPredicate = ("suppkey" IN (BIGINT '2', BIGINT '3')), dynamicFilters = {"suppkey" = #df_762}]
     │      Layout: [suppkey:bigint, $hashvalue:bigint]
     │      Estimates: {rows: 60175 (1.03MB), cpu: 528.88k, memory: 0B, network: 0B}/{rows: 1204 (21.16kB), cpu: 1.03M, memory: 0B, network: 0B}/{rows: 1204 (21.16kB), cpu: 1.05M, memory: 0B, network: 0B}
     │      CPU: 44.00ms (10.48%), Scheduled: 94.00ms (6.47%), Output: 1140 rows (20.04kB)
     │      Input avg.: 60175.00 rows, Input std.dev.: 0.00%
     │      $hashvalue := combine_hash(bigint '0', COALESCE("$operator$hash_code"("suppkey"), 0))
     │      suppkey := suppkey:bigint:REGULAR
     │      Input: 60175 rows (528.88kB), Filtered: 98.11%
     │      Dynamic filters: {[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]}
```

Reply:
Imagine you have 8 joins (which is not that uncommon in tpc-h/tpc-ds suite), then that explain would be:
```
ScanFilterProject[table = hive:default:lineitem, grouped = false, filterPredicate = ("suppkey" IN (BIGINT '2', BIGINT '3')), dynamicFilters = {"suppkey" = #df_762}]
     │      Layout: [suppkey:bigint, $hashvalue:bigint]
     │      Estimates: {rows: 60175 (1.03MB), cpu: 528.88k, memory: 0B, network: 0B}/{rows: 1204 (21.16kB), cpu: 1.03M, memory: 0B, network: 0B}/{rows: 1204 (21.16kB), cpu: 1.05M, memory: 0B, network: 0B}
     │      CPU: 44.00ms (10.48%), Scheduled: 94.00ms (6.47%), Output: 1140 rows (20.04kB)
     │      Input avg.: 60175.00 rows, Input std.dev.: 0.00%
     │      $hashvalue := combine_hash(bigint '0', COALESCE("$operator$hash_code"("suppkey"), 0))
     │      suppkey := suppkey:bigint:REGULAR
     │      Input: 60175 rows (528.88kB), Filtered: 98.11%
     │      Dynamic filters: {[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]},{[id=df_762, SortedRangeSet[type=bigint, ranges=1, [2]], collection time=3.59s]}
```
which is really hard to read.
We should print dynamic filters in separate lines for sake of readability. Hence `VERBOSE` comes useful.

>  Even with multiple DFs per FilterNode it should be comparable or less than Estimates printing

Yeah, that's not great either as well as long `filterPredicate` and `dynamicFilters`. Super long lines are hard to read.

Reply:
I've updated this to
```
ScanFilterProject[table = hive:default:lineitem, grouped = false, filterPredicate = true, dynamicFilters = {"orderkey" = #df_458, "orderkey" = #df_459}]
     │  │      Layout: [orderkey:bigint, $hashvalue:bigint]
     │  │      Estimates: {rows: 60175 (1.03MB), cpu: 528.88k, memory: 0B, network: 0B}/{rows: 60175 (1.03MB), cpu: 1.03M, memory: 0B, network: 0B}/{rows: 60175 (1.03MB), cpu: 2.07M, memory: 0B, network: 0B}
     │  │      CPU: 11.00ms (13.10%), Scheduled: 63.00ms (24.42%), Output: 10000 rows (175.78kB)
     │  │      Input avg.: 10000.00 rows, Input std.dev.: 0.00%
     │  │      $hashvalue := combine_hash(bigint '0', COALESCE("$operator$hash_code"("orderkey"), 0))
     │  │      orderkey := orderkey:bigint:REGULAR
     │  │      Input: 10000 rows (87.89kB), Filtered: 0.00%
     │  │      Dynamic filters:
     │  │          :: id=df_458, [ SortedRangeSet[type=bigint, ranges=2000, {[1], ..., [2000]}] ], collection time=105.42ms
     │  │          :: id=df_459, ALL, collection time=87.87ms
```
I think this is information is useful and formatted compactly enough to not be hidden behind VERBOSE flag.

Reply:
> I think this is information is useful and formatted compactly enough to not be hidden behind VERBOSE flag.

It's more of a question how large we want `EXPLAIN ANALYZE` to be? I'm fine keeping it non-verbose, although I think it will make (some) `EXPLAINS` much larger. @findepi wdyt?

Reply:
>     │  │      Dynamic filters:
>     │  │          :: id=df_458, [ SortedRangeSet[type=bigint, ranges=2000, {[1], ..., [2000]}] ], collection time=105.42ms
>     │  │          :: id=df_459, ALL, collection time=87.87ms

i like the output. It seems you can afford limit of 3-5 here as well.
I would only suggest different formatting, perhaps change
`:: id=df_459, {details}` to
`- df_459: {details}`

Reply:
Updated to
```
Dynamic filters:
     │  │          - df_458, [ SortedRangeSet[type=bigint, ranges=2000, {[1], ..., [2000]}] ], collection time=1.14s
     │  │          - df_459, ALL, collection time=1.12s
```
I'm being conservative about the limit for the case where we could have long string values.

Reply:
```
[ SortedRangeSet[type=bigint, ranges=2000, {[1], ..., [2000]}] ]
```
Do we need extra outer `[]` here? Could we remove them?

Reply:
The outer `[]` can have a NULL in addition to the SortedRangeSet.
We could remove it when we know that NULL is not allowed, but I kept it for the sake of consistency